pkg/runtime/grpc/oci.go

```diff
@@ -128,21 +138,40 @@ func (r *Runtime) runGadget(gadgetCtx runtime.GadgetContext, target target, allP
 	defer conn.Close()
 	client := api.NewGadgetManagerClient(conn)
 
-	runRequest := &api.GadgetRunRequest{
-		ImageName:   gadgetCtx.ImageName(),
-		ParamValues: allParams,
-		Args:        gadgetCtx.Args(),
-		LogLevel:    uint32(gadgetCtx.Logger().GetLevel()),
-		Timeout:     int64(gadgetCtx.Timeout()),
-		Version:     api.VersionGadgetRunProtocol,
-	}
-
 	runClient, err := client.RunGadget(connCtx)
 	if err != nil && !errors.Is(err, context.Canceled) {
 		return nil, err
 	}
 
-	controlRequest := &api.GadgetControlRequest{Event: &api.GadgetControlRequest_RunRequest{RunRequest: runRequest}}
+	var controlRequest *api.GadgetControlRequest
+
+	interactive := true
+	if gadgetCtx.UseInstance() {
+		gadgetCtx.Logger().Debugf("attaching to gadget instance %s", gadgetCtx.ImageName())
+		controlRequest = &api.GadgetControlRequest{
+			Event: &api.GadgetControlRequest_AttachRequest{
+				AttachRequest: &api.GadgetAttachRequest{
+					Id:      gadgetCtx.ImageName(),
```

Comment:
Yeah, we could make it `ImageNameOrID()` (coming with many changes across the repo) or just add a `ID()` that will also read the same underlying field just to be semantically correct - both feel weird somehow.

Maybe we should've added `WithImageName()` / `WithInstanceID()` (and error out if none is set), instead of making the imagename a first-class citizen 😄